doc/how-to-run-in-windows.rst

```diff
@@ -0,0 +1,352 @@
+will be in scope, so configuring packages should be able to find them.
+
+Using Cabal on Windows
+----------------------
+
+Cabal will also be installed automatically by GHCup. Once the location of the
+binary is present in the ``PATH``, ``cabal`` can be invoked from any Windows
+shell, just as GHC.
+
+::
+
+   > cabal --help
+   Command line interface to the Haskell Cabal infrastructure.
+
+   See http://www.haskell.org/cabal/ for more information.
+
+   Usage: cabal-3.10.2.1.exe [GLOBAL FLAGS] [COMMAND [FLAGS]]
+
+   ...
+
+As both ``cabal.exe`` and ``ghc.exe`` are placed in the same location by GHCup,
+normally cabal will be able to invoke GHC as said location will be in the
+``PATH``.
+
+All the above extra configurations will also be needed when using ``cabal``, and
+one can provide all of them via the following local project file:
+
+::
+
+   msys2> cat << EOF > cabal.project.local
+   package *
+     extra-include-dirs: C:\msys64\ucrt64\include
+     extra-lib-dirs: C:\msys64\ucrt64\lib
+     extra-prog-path: C:\msys64\ucrt64\bin,
+                      C:\msys64\usr\bin
+   EOF
+
+Note that ``extra-include-dirs: x`` translates to the GHC option ``-Ix``,
+``extra-lib-dirs: x`` translates to the GHC option ``-Lx`` and
+``extra-prog-path: x`` results in the path ``x`` being added to the ``PATH``
+environment when calling an executable by cabal.
+
+However, this has the inconvenience that we would need to set this up for every
+project that we want to build, and we sometimes wouldn't be able to install
+binaries from Hackage directly as there is no local project to refer to.
+
+The recommended way of setting this is then modifying the global cabal
+configuration, whose location you can find by calling ``cabal --help``:
+
+::
+
+   > cabal --help
+   ...
+   You can edit the cabal configuration file to set defaults:
+      C:\Users\<YourName>\AppData\Roaming\cabal\config
+
+In there, just add the lines mentioned above. For even better user experience,
+it is advisable to add also two additional directories to the
+``extra-prog-path`` section:
+
+- ``C:\ghcup\bin``: in the unexpected case that said directory is not in the
+  ``PATH`` already (which would mean that ``cabal.exe`` was invoked via its full
+  path or that it is not in the standard location).
+
+- ``C:\Users\<YourName>\AppData\Roaming\cabal\bin`` or whichever directory cabal
+  installs binaries, so that other Haskell executables are also findable by
+  cabal-invoked tools. See whether the ``installdir:`` section is uncommented
+  and use its value.
+
+The configuration would then look like this:
+
+::
+
+   > cat C:\\Users\\<YourName>\\AppData\\Roaming\\cabal\\config
+   ...
+   extra-include-dirs: C:\msys64\ucrt64\include
+   ...
+   extra-lib-dirs: C:\msys64\ucrt64\lib
+   ...
+   extra-prog-path: C:\ghcup\bin,
+                    C:\Users\<YourName>\AppData\Roaming\cabal\bin,
+                    C:\msys64\ucrt64\bin,
+                    C:\msys64\usr\bin
+   ...

```

Comment:
Why isn't ghcup getting this right for us?

Reply:
It is. But this shouldn't be part of the cabal documentation.

Reply:
GHCup is setting it for us. I hope I made it clear in the new version.

If this should not be part of the cabal documentation, then what should be part of it? I don't understand your point.